tests/system_tests_multicast.py

```diff
@@ -429,6 +438,137 @@ def test_999_check_for_leaks(self):
+        #                      +---------+
+        #
+        # Each router has 2 multicast consumers
+        # INT.A has a multicast sender
+
+        cls.routers = []
+
+        interrouter_port_B_to_A = cls.tester.get_port()
+        interrouter_port_C_to_A = cls.tester.get_port()
+        interrouter_port_C_to_B = cls.tester.get_port()
+
+        router('INT.A', 'interior',
+               [('listener', {'role': 'inter-router',
+                              'port': interrouter_port_B_to_A}),
+                ('listener', {'role': 'inter-router',
+                              'port': interrouter_port_C_to_A})])
+        cls.INT_A = cls.routers[0]
+        cls.INT_A.listener = cls.INT_A.addresses[0]
+
+        router('INT.B', 'interior',
+               [('connector', {'name': 'connectorToA',
+                               'role': 'inter-router',
+                               'port': interrouter_port_B_to_A}),
+                ('listener', {'role': 'inter-router',
+                              'port': interrouter_port_C_to_B})])
+        cls.INT_B = cls.routers[1]
+        cls.INT_B.listener = cls.INT_B.addresses[0]
+
+        router('INT.C', 'interior',
+               [('connector', {'name': 'connectorToA',
+                               'role': 'inter-router',
+                               'port': interrouter_port_C_to_A}),
+                ('connector', {'name': 'connectorToB',
+                               'role': 'inter-router',
+                               'port': interrouter_port_C_to_B})])
+        cls.INT_C = cls.routers[2]
+        cls.INT_C.listener = cls.INT_C.addresses[0]
+
+        cls.INT_A.wait_router_connected('INT.B')
+        cls.INT_A.wait_router_connected('INT.C')
+        cls.INT_B.wait_router_connected('INT.A')
+        cls.INT_B.wait_router_connected('INT.C')
+        cls.INT_C.wait_router_connected('INT.A')
+        cls.INT_C.wait_router_connected('INT.B')
+
+        # Client topology:
+        # all routes have 2 receivers
+        # Interior INT_A has 1 sender
+        #
+        cls.config = [
+            # Interior router INT_A:
+            {'router':      cls.INT_A,
+             'senders':     ['S-INT_A-1'],
+             'receivers':   ['R-INT_A-1', 'R-INT_A-2'],
+             'subscribers': 2,
+             'remotes':     2,
+             },
+            # Interior router INT_B:
+            {'router':      cls.INT_B,
+             'senders':     [],
+             'receivers':   ['R-INT_B-1', 'R-INT_B-2'],
+             'subscribers': 2,
+             'remotes':     2,
+             },
+            # Interior router INT_C:
+            {'router':      cls.INT_C,
+             'senders':     [],
+             'receivers':   ['R-INT_C-1', 'R-INT_C-2'],
+             'subscribers': 2,
+             'remotes':     2,
+             }
+        ]
+
+    #
+    # run any negative tests first so that if we screw up the internal
+    # state of the brokers the positive tests will likely fail
```

Comment:
Should this comment be removed ? What are the negative tests ?

Reply:
Removed comment.